# Amharic News Dataset EDA

In [1]:
#import library
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Fetch the Data

In [2]:
data = pd.read_csv('../data/Amharic News Dataset.csv')

In [3]:
data.head()

headline category               date  \
0  የኦሊምፒክ ማጣሪያ ተሳታፊዎች የሚለዩበት ቻምፒዮና እየተካሄደ ይገኛል     ስፖርት   January 14, 2021   
1                                   አዲስ ዘመን ድሮ     መዝናኛ  December 28, 2020   
2             የአረንጓዴ ጎርፍ በጎ አድራጎት አምባሳደሮች ተሰየሙ     ስፖርት    January 6, 2021   
3        የሊጉ በቢዝነስ ሞዴል መመራት አበረታች ጅምር መሆኑ ተገለጸ     ስፖርት    January 6, 2021   
4    የኦሊምፒክ ሥራ አስፈፃሚው እስከ ቶኪዮ ኦሊምፒክ ማግስት ይቀጥላል     ስፖርት    January 6, 2021   

  views                                            article  \
0     2  ብርሃን ፈይሳየኢትዮጵያ ቦክስ ፌዴሬሽን በየዓመቱ የሚያዘጋጀው የክለቦች ቻ...   
1     4   የአዲስ ዘመን ጋዜጣ ቀደምት ዘገባዎች በእጅጉ ተነባቢ ዛሬም ላገኛቸው በ...   
2     6  ቦጋለ አበበየአዲስ አበባ ከተማ አስተዳደር ስፖርት ኮሚሽን ከኢትዮጵያ አረ...   
3     5  ብርሃን ፈይሳአዲስ አበባ፡- የኢትዮጵያ ፕሪምየር ሊግ በሼር ካምፓኒ እንዲ...   
4    12  ቦጋለ አበበ የኢትዮጵያ ኦሊምፒክ ኮሚቴ አርባ አምስተኛ መደበኛ ጠቅላላ ጉ...   

                                link  
0  https://www.press.et/Ama/?p=39481  
1  https://www.press.et/Ama/?p=38334  
2  https://www.press.et/Ama/?p=39010  
3  https://www.press.et/Ama/?p=39011  
4  https://www.press.et/Ama/?p=39012

In [10]:
data.shape

(51483, 6)

There are 51483 datasets and 6 columns namely headline, category, date, views, article and link

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51483 entries, 0 to 51482
Data columns (total 6 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   headline  51470 non-null  object
 1   category  51482 non-null  object
 2   date      51483 non-null  object
 3   views     51483 non-null  object
 4   article   51483 non-null  object
 5   link      51483 non-null  object
dtypes: object(6)
memory usage: 2.4+ MB


In [5]:
data.groupby('category').count().sort_values(by='views',ascending=False)

headline   date  views  article   link
category                                          
ሀገር አቀፍ ዜና     20665  20674  20674    20674  20674
ስፖርት           10411  10411  10411    10411  10411
ፖለቲካ            9325   9325   9325     9325   9325
ዓለም አቀፍ ዜና      6543   6543   6543     6543   6543
ቢዝነስ            3894   3894   3894     3894   3894
መዝናኛ             632    635    635      635    635

ሀገር አቀፍ ዜና meaning Domestic News is the largest category followed by ስፖርት meaing Sport is the second largest catefory and መዝናኛ meaning Entertainment has the least amount of articles.

In [17]:
data.groupby('category')['views'].sum()

category
ሀገር አቀፍ ዜና    3274810224281272816173311154641105033121819151...
መዝናኛ          4244420325553633703961542127263068492627460333...
ስፖርት          2651210311108779710910101213791681310231059892...
ቢዝነስ          6098021090108612001330139114291227252327599424...
ዓለም አቀፍ ዜና    8982495231016554331158010657546427111793698918...
ፖለቲካ          3071863703582643952262633192522354052351161213...
Name: views, dtype: object

ቢዝነስ meaning Business has the highest number of views while ዓለም አቀፍ ዜና meaning International News has the lowest number of views.

In [7]:
data[data['views']=='Unknown'].groupby('category').count()

headline  date  views  article  link
category                                        
ሀገር አቀፍ ዜና      6743  6743   6743     6743  6743
መዝናኛ               5     5      5        5     5
ስፖርት            9270  9270   9270     9270  9270
ቢዝነስ            3198  3198   3198     3198  3198
ዓለም አቀፍ ዜና      4514  4514   4514     4514  4514
ፖለቲካ            9212  9212   9212     9212  9212

In [8]:
data[data['views']!='Unknown'].groupby('category').count()

headline   date  views  article   link
category                                          
ሀገር አቀፍ ዜና     13922  13931  13931    13931  13931
መዝናኛ             627    630    630      630    630
ስፖርት            1141   1141   1141     1141   1141
ቢዝነስ             696    696    696      696    696
ዓለም አቀፍ ዜና      2029   2029   2029     2029   2029
ፖለቲካ             113    113    113      113    113

## Article Length

In [11]:
data['article_length'] = data['article'].str.split().str.len()

In [12]:
data.head()

headline category               date  \
0  የኦሊምፒክ ማጣሪያ ተሳታፊዎች የሚለዩበት ቻምፒዮና እየተካሄደ ይገኛል     ስፖርት   January 14, 2021   
1                                   አዲስ ዘመን ድሮ     መዝናኛ  December 28, 2020   
2             የአረንጓዴ ጎርፍ በጎ አድራጎት አምባሳደሮች ተሰየሙ     ስፖርት    January 6, 2021   
3        የሊጉ በቢዝነስ ሞዴል መመራት አበረታች ጅምር መሆኑ ተገለጸ     ስፖርት    January 6, 2021   
4    የኦሊምፒክ ሥራ አስፈፃሚው እስከ ቶኪዮ ኦሊምፒክ ማግስት ይቀጥላል     ስፖርት    January 6, 2021   

  views                                            article  \
0     2  ብርሃን ፈይሳየኢትዮጵያ ቦክስ ፌዴሬሽን በየዓመቱ የሚያዘጋጀው የክለቦች ቻ...   
1     4   የአዲስ ዘመን ጋዜጣ ቀደምት ዘገባዎች በእጅጉ ተነባቢ ዛሬም ላገኛቸው በ...   
2     6  ቦጋለ አበበየአዲስ አበባ ከተማ አስተዳደር ስፖርት ኮሚሽን ከኢትዮጵያ አረ...   
3     5  ብርሃን ፈይሳአዲስ አበባ፡- የኢትዮጵያ ፕሪምየር ሊግ በሼር ካምፓኒ እንዲ...   
4    12  ቦጋለ አበበ የኢትዮጵያ ኦሊምፒክ ኮሚቴ አርባ አምስተኛ መደበኛ ጠቅላላ ጉ...   

                                link  article_length  
0  https://www.press.et/Ama/?p=39481             298  
1  https://www.press.et/Ama/?p=38334             475  
2  https://www.press.et/Ama/?p=39010             265  
3  https://www.press.et/Ama/?p=39011             296  
4  https://www.press.et/Ama/?p=39012             476

In [13]:
data.sort_values(by='article_length',ascending=False)

headline    category  \
22598  ‹‹ከእኔ ክስና እስራት በስተጀርባ በመንግሥት ውስጥ ጡንቻቸው የጠነከረ ማ...        ፖለቲካ   
23909            ‹‹ኢትዮጵያ የባህር ኃይሏን መልሳ ለመመሥረት ማጤን አለባት››        ፖለቲካ   
8100                   የውጪ ጉዲፈቻ የተከለከለበት ምክንያትና አስከልካዮቹ!        ፖለቲካ   
7395                                 ኧርያል ሻሮንና አያት አክራስ   ዓለም አቀፍ ዜና   
6281   ወደ አገሩ ሲመለስ “አዲስ አበባ አሸነፈች፤ ቺካጐ ተሸነፈች” የተባለለት ...        ቢዝነስ   
...                                                  ...         ...   
35562                                       ቻምፒዮኖቹ ሲገመገሙ        ስፖርት   
35563               ፕሪምየር ሊጉ ዛሬ እና ነገ በሚደረጉ ጨዋታዎች ይጠናቀቃል        ስፖርት   
48388   ሙስሊሙ ህብረተሰብ እራሱን ከጽንፈኞችና አክራሪዎች መጠበቅ እንዳለበት ተገለጸ        ፖለቲካ   
35564                ከ20 አመት በታች ቡድኑ ነገ የመልስ ጨዋታውን ያደርጋል        ስፖርት   
35882    ጉዳይ-ኦን - አገልግሎት ሰጪዎችን በቴክኖሎጂ የሚያግዝ የሞባይል መተግበሪያ  ሀገር አቀፍ ዜና   

                                   date    views  \
22598                      29 July 2018  Unknown   
23909                       9 July 2017  Unknown   
8100                         2021-01-04      672   
7395   Saturday, 13 October 2012 13:46      2647   
6281    Saturday, 08 August 2015 09:40     10596   
...                                 ...      ...   
35562                     June 21, 2014  Unknown   
35563                     June 21, 2014  Unknown   
48388                   August 16, 2013  Unknown   
35564                      May 24, 2014  Unknown   
35882                  October 21, 2020  Unknown   

                                                 article  \
22598  አቶ መላኩ ፈንታ፣ የቀድሞ ገቢዎችና ጉምሩክ ባለሥልጣን ዋና ዳይሬክተርየዛ...   
23909  አቶ አብዱል መሐመድ፣ የኢንተርአፍሪካ ግሩፕ መሥራች በትክክል እንደተረዳኸ...   
8100   ጥር 20/2012 ለህትመት የበቃችውን አዲስ ማለዳ ጋዜጣ በመጥቀስና በእለ...   
7395   በእርግጥም ይስሀቅ ራቢን በህይወት የኖሩበት እያንዳንዱ አመት በታላቅ ታሪ...   
6281     በ15 ዓመት 30 ት/ቤቶች አሠርቶ፣ ከ225 ሺህ ዜጐች በላይ አስተምሯ...   
...                                                  ...   
35562                                                      
35563                                                      
48388                                                      
35564                                                      
35882                                                 \n   

                                                    link  article_length  
22598    https://www.ethiopianreporter.com/article/12186            6738  
23909  https://www.ethiopianreporter.com/content/%E2%...            3504  
8100              https://addismaleda.com/archives/10141            3327  
7395   https://www.addisadmassnews.com/index.php?opti...            3300  
6281   https://www.addisadmassnews.com/index.php?opti...            3282  
...                                                  ...             ...  
35562            https://soccerethiopia.net/football/302               0  
35563            https://soccerethiopia.net/football/303               0  
48388                    https://waltainfo.com/am/28030/               0  
35564            https://soccerethiopia.net/football/281               0  
35882  https://amharic.voanews.com//a/%E1%8C%89%E1%8B...               0  

[51483 rows x 7 columns]